ocs_ci/ocs/pgsql.py

```diff
@@ -0,0 +1,293 @@
+            else:
+                return False
+            break
+
+    def get_pgbench_status(self, pgbench_pod_name):
+        """
+        Get pgbench status
+
+        Args:
+            pgbench_pod_name (str): Name of the pgbench pod
+
+        Returns:
+            str: state of pgbench pod (running/completed)
+
+        """
+        pod_obj = get_pod_obj(pgbench_pod_name, namespace=RIPSAW_NAMESPACE)
+        status = pod_obj.get().get(
+            'status'
+        ).get('containerStatuses')[0].get('state')
+        if list(status.keys())[0] == 'running':
+            return 'running'
+        elif list(status.keys())[0] == 'terminated':
+            return status['terminated']['reason']
+
+    def wait_for_pgbench_status(self, status, timeout=None):
+        """
+        Wait for pgbench benchmark pods status to reach running/completed
+
+        Args:
+            status (str): status to reach Running or Completed
+            timeout (int): Time in seconds to wait
+
+        """
+
+        timeout = timeout if timeout else 43200
+        # Wait for pg_bench pods to initialized and running
+        log.info(f"Waiting for pgbench pods to be reach {status} state")
+        pgbench_pod_objs = self.get_pgbench_pods()
+        for pgbench_pod_obj in pgbench_pod_objs:
+            wait_for_resource_state(
+                resource=pgbench_pod_obj, state=status, timeout=timeout
+            )
+
+    def validate_pgbench_run(self, pgbench_pods):
+        """
+        Validate pgbench run
+
+        Args:
+            pgbench pods (list): List of pgbench pods
+
+        Returns:
+            pg_output (list): pgbench outputs in list
+
+        """
+        pg_output = []
+        for pgbench_pod in pgbench_pods:
+            log.info(f"pgbench_client_pod===={pgbench_pod.name}====")
+            output = run_cmd(f'oc logs {pgbench_pod.name}')
+            pg_output = utils.parse_pgsql_logs(output)
+            # list_pg_output.append(pg_output)
+            log.info(
+                "*******PGBench output log*********\n"
+                f"{pg_output}"
+            )
+
+        # for data in list_pg_output:
+        for data in pg_output:
+            latency_avg = data['latency_avg']
+            if not latency_avg:
+                raise UnexpectedBehaviour(
+                    "PGBench failed to run, "
+                    "no data found on latency_avg"
+                )
+        log.info("PGBench completed successfully")
+        return pg_output
+
+    def collect_data_to_googlesheet(self, pg_output, sheet_name, sheet_index):
+        """
+        Collect pgbench output to google spread sheet
+
+        Args:
+            pg_output (list):  pgbench outputs in list
+            sheet_name (str): Name of the sheet
+            sheet_index (int): Index of sheet
+
+        """
+        # Collect data and export to Google doc spreadsheet
+        g_sheet = GoogleSpreadSheetAPI(sheet_name, sheet_index)
+        for lat in pg_output:
```

Comment:
we should append replica count and num_clients as well, since that is configurable now.

Reply:
utils.parse_pgsql_logs(output) fails to get pg_output for multiple pgbench pods. 
Working on fixing that one.

Reply:
I have fixed utils.parse_pgsql_logs(output)  to return pg_data from multiple pgbench pods.
Also, removed collecting the data to google sheet as we are not collecting perf details more over just adding the pg_out.
Please let me know otherwise.